test/test_pyroapi.py

```diff
@@ -0,0 +1,11 @@
+from pyroapi import handlers, pyro_backend
+from pyroapi.tests import *  # noqa F401
+
+pytestmark = pytest.mark.filterwarnings("ignore::numpyro.compat.util.UnsupportedAPIWarning")
+
+
+@pytest.yield_fixture
+def backend():
+    with pyro_backend('numpy'):
+        with handlers.seed(rng_seed=1):
```

Comment:
Hmm, I think we should try to avoid any backend-specific code here. I have a few questions:
1. Why is this needed here, even though we don't have this in pyroapi/test/test_tests.py?
2. Can we add a hook to the pyro_backend context manager so that, for each backend a set of contexts is entered in pyro_backend? That way we can set a default seed when entering a `with pyro_backend('numpy')` context?

Reply:
@neerajprad does this solution seem to hacky?
https://github.com/pyro-ppl/pyro-api/pull/new/set-seed-by-default

Reply:
1. All inference algorithms require an explicit `rng_seed` in NumPyro, which is used for sampling from the models, etc. The tests in `test_tests.py` are only currently testing functionality and therefore don't need to fix the rng seed (for other backends). If we ever fix the seed, this would be a backend-agnostic way of doing it. An alternative would be to sample a `random.randint` in the compatibility wrapper as a seed if we aren't inside the `seed` handler context. WDYT?
2. We can do that, but we can only do so for the known backends, or provide extra arguments so that new backends can similarly register context managers to be entered into. Also, this will have a different behavior since the tests will be deterministic for NumPyro (also a problem with using handlers like above) unlike other backends, so using a global non-deterministic seed seems like a better solution.

> @neerajprad does this solution seem to hacky?
https://github.com/pyro-ppl/pyro-api/pull/new/set-seed-by-default

The only issue that I see there is that it will fix the seed by default for all tests? Does that seem fine to you?

Reply:
The good thing with your proposal is that is guaranteed to work for other cases too, e.g. when sampling from a model etc. outside an inference context.

Reply:
I commented on your proposal. I think that is doable if we change the behavior of nested seed handlers in NumPyro.